datafusion/functions-aggregate-common/src/aggregate/groups_accumulator.rs

```diff
@@ -427,10 +427,10 @@ impl GroupsAccumulator for GroupsAccumulatorAdapter {
         true
     }
 
-    fn with_group_indices_order_mode(
-        mut self: Box<Self>,
+    fn set_group_indices_order_mode(
+        &mut self,
         group_indices_order_mode: &InputOrderMode,
-    ) -> Result<Box<dyn GroupsAccumulator>> {
+    ) -> Result<()> {
         if !self.states.is_empty() {
```

Comment:
this function is basically just modifying `mut self` so I think it makes it easier to see if the API allows modification directly

Reply:
I wanted to create in a different PR a specialized implementation here to avoid the need for condition checking each batch and also have optimized structure by only having 1 accumulator at any given moment as we only need to save the current group

Reply:
Will create a new branch with what I mean

Reply:
Here I created a group accumulator for ArrayAgg and specilized accumulator for ordering:

https://github.com/rluvaton/datafusion/pull/2